pkg/rancher-desktop/main/diagnostics/wslFromStore.ts

```diff
@@ -0,0 +1,221 @@
+import os from 'os';
+import path from 'path';
+
+import { DiagnosticsCategory, DiagnosticsChecker } from './types';
+
+import { spawnFile } from '@pkg/utils/childProcess';
+import Logging from '@pkg/utils/logging';
+
+const console = Logging.diagnostics;
+
+/**
+ * Check if WSL was installed from Microsoft Store.
+ */
+class CheckWSLFromStore implements DiagnosticsChecker {
+  readonly id = 'WSL_FROM_STORE';
+
+  category = DiagnosticsCategory.Testing;
+  applicable(): Promise<boolean> {
+    return Promise.resolve(process.platform === 'win32');
+  }
+
+  get system32() {
+    return path.join(process.env['SystemRoot'] ?? 'C:\\Windows', 'System32');
+  }
+
+  get powerShellExecutable(): string {
+    return path.join(this.system32, 'WindowsPowerShell', 'v1.0', 'powershell.exe');
+  }
+
+  /**
+   * Check if the current version of Windows can use WSL from the store.
+   */
+  async canUseStoreWSL(): Promise<boolean> {
+    const [major, minor, build] = os.version().split('.').map(v => parseInt(v, 10));
+
+    if (major < 10) {
+      // Windows 8.x or lower?
+      return false;
+    }
```

Comment:
I thought Rancher Desktop requires Windows 10 or later to run at all and #1294 will prevent the user from even starting it on Windows 8 or earlier.

Reply:
Yes, this is just covering all cases; it should never be hit in practice. Hence the `?` there; perhaps I should switch it to `‽` to properly express the incredulity?